v2/cli/init_command_test.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/brigadecore/brigade/v2/internal/file"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNormalizeLanguage(t *testing.T) {
+	testCases := []struct {
+		name     string
+		expected string
+	}{
+		{"typescript", "ts"},
+		{"ts", "ts"},
+		{"javascript", "js"},
+		{"js", "js"},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			result, err := normalizeLanguage(testCase.name)
+			require.NoError(t, err)
+			require.Equal(t, result, testCase.expected)
+		})
+	}
+}
+
+func TestIDMatch(t *testing.T) {
+	testCases := []struct {
+		id string
+	}{
+		{"abc"},
+		{"a11"},
+		{"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.id, func(t *testing.T) {
+			err := idMatch(testCase.id)
+			require.NoError(t, err)
+		})
+	}
+
+	wrongTestCases := []struct {
+		id string
+	}{
+		{"a"},
+		{"aa"},
+		{"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"},
+		{"1aa"},
+		{"Hello"},
+	}
+
+	for _, testCase := range wrongTestCases {
+		t.Run(testCase.id, func(t *testing.T) {
+			err := idMatch(testCase.id)
+			require.Errorf(t, err, "invalid value %q for --id flag"+
+				"(must be between 3 and 63 characters, first char must be"+
+				" a letter)", testCase.id)
+		})
+	}
+}
+
+func TestAddToGitignore(t *testing.T) {
+	err := addToGitignore(".brigade/secrets.yaml")
+	require.NoError(t, err)
+
+	// Verify .gitignore is created
+	fileExists, err := file.Exists(".gitignore")
+	require.True(t, fileExists)
+	require.NoError(t, err)
+
+	// Verify .gitignore contains .brigade/secrets.yaml
+	verifyContents(t, ".gitignore", ".brigade/secrets.yaml")
+
+	// Clear .gitignore, test functionality of writing to existing .gitignore
+	err = ioutil.WriteFile(".gitignore", []byte(``), 0644)
+	require.NoError(t, err)
+	err = addToGitignore(".brigade/secrets.yaml")
+	require.NoError(t, err)
+	verifyContents(t, ".gitignore", ".brigade/secrets.yaml")
+
+	// Remove the created .gitignore file
+	err = os.Remove(".gitignore")
+	require.NoError(t, err)
+}
```

Comment:
I think this test case ends up modifying the project's own files which can end up being problematic in unexpected ways down the road.

We can pair together today on how to test this functionality _without_ adding / modifying / deleting the project's own files.

Reply:
Well... you just taught me something cause I've used tempdir for this stuff often and never realized T had any built in support for that. Neat.